### DISCUSSION TITLE
Daylight saving time

### DISCUSSION THREAD

**Discussion by 098tomtom** [simpleavr/s14clock#16] [Ideas]
Any interest in adding automatic daylight saving time adjustment?

**Comment by simpleavr** [comment]
Thanks for your idea. Will have to look at it later this month as I am engage in another project right now.

**Comment by simpleavr** [comment]
FW2.03 avaliable for download / install. This firmware allows the use of proper time zone configuration via POSIX TZ strings. Setup page will use client browser's geo-location capability to "guess" the correct setting. System does not assume an always correct setup, user has to click the "USE" button to set TZ string up, as the code relies on an external repository for information and it may fail in the future.

Please report any issue via this thread.

@098tomtom DST now available.

**Reply by 098tomtom** [reply]
Ripper! Thanks for your continued support. It's such a great little clock. Thanks again

**Comment by 098tomtom** [comment]
I've noticed in the 8th field where @xyz can be used for TZ... The field isn't large enough to capture the string when you click "use". My local time is showing one hour behind and I'm thinking this may be the reason? It's correctly identified my region/locale but on the wrong time.

@AEST-10AEDT,M4.1.0,M10.1.0/3 this is what shows at the top of the interface

@AEST-10AEDT,M4. is all that's recorded in the 8th field

Any ideas?
Thanks!

**Reply by simpleavr** [reply]
Thanks for your feedback, I haven't really been careful and look at more regions's configuration.

You are correct, the field does not accomodate longer TZ strings with 24 characters.
I was trying to avoid changing the data structure too much and re-used the "addn text" field 8.
Looks like adding a dedicated field would be the correct way to do this and I will work on it.
Will update this thread when it's ready for testing. Thanks for your testing and report.

**Comment by simpleavr** [comment]
I made code change and now moved the alternate NTP server and Posix TZ configurations from the "contents" slot 8 to new storage. So now max 40 characters for both configurations, do not use the '@' prefix for these configurations.
Please try and report if there are any issues.
V2.03 binaries available in the firmware install page.

**Reply by 098tomtom** [reply]
Brilliant! Works perfectly. Hats off to ya. Thank you
